src/mumble/pluginAPI/Plugin.h

```diff
@@ -0,0 +1,155 @@
+	/// @param mumbleAPIVersion The Version of the plugin-API the Mumble client runs with
+	/// @param minimalExpectedAPIVersion The minimal Version the Mumble clients expects this plugin to meet in order to load it
+	PLUGIN_EXPORT void setMumbleInfo(Version_t mumbleVersion, Version_t mumbleAPIVersion, Version_t minimalExpectedAPIVersion);
+
+	// functions for general plugin info
+	
+	/// Gets the name of the plugin.
+	///
+	/// @param[out] nameBuffer The buffer into which the name of the plugin shall be copied by this function
+	/// @param bufferSize The size of the buffer - The length of the name of the plugin (including the terminating zero-byte) must not exceed
+	/// 	this length
+	PLUGIN_EXPORT void getName(char *nameBuffer, size_t bufferSize);
+
+	/// Gets the plugin's version in a display-ready representation.
+	///
+	/// @param[out] versionBuffer The buffer into which the version string shall be copied by this function
+	/// @param bufferSize The size of the buffer - The length of the version (including terminating zero-byte) must not exceed this length
+	PLUGIN_EXPORT void getDisplayVersion(char *versionBuffer, size_t bufferSize);
+
+	/// Gets the Version of this plugin
+	///
+	/// @returns The plugin's version
+	PLUGIN_EXPORT Version_t getVersion();
+
+	/// Gets the Version of the plugin-API this plugin intends to use.
+	/// Mumble will decide whether this plugin is loadable or not based on the return value of this function.
+	///
+	/// @return The respective API Version
+	PLUGIN_EXPORT Version_t getApiVersion();
+
+	/// Gets the name of the plugin author(s)
+	///
+	/// @param[out] authorBuffer The buffer into which the author name shall be copied by this function
+	/// @param bufferSize The size of the buffer - The length of the author name (including the terminating zero-byte) must not exceed
+	/// 	this length.
+	PLUGIN_EXPORT void getAuthor(char *authorBuffer, size_t bufferSize);
+
+	/// Gets the description of the plugin
+	///
+	/// @param[out] descriptionBuffer The buffer into which the description shall be copied by this function
+	/// @param bufferSize The siue of the buffer - The length of the description (including the terminating zero-byte) must not exceed
+	/// 	this length.
+	PLUGIN_EXPORT void getDescription(char *descriptionBuffer, size_t bufferSize);
+
+	/// Provides the MumbleAPI struct to the plugin. This struct contains function pointers that can be used
+	/// to interact with the Mumble client. It is up to the plugin to store this struct somewhere if it wants to make use
+	/// of it at some point.
+	///
+	/// @param api The MumbleAPI struct
+	PLUGIN_EXPORT void setAPIFunctions(const struct MumbleAPI *api);
+
+
+
+	/// Sets the ID of this plugin. This is the ID Mumble will reference this plugin with and by which this plugin
+	/// can identify itself when communicating with Mumble.
+	///
+	/// @param id The ID for this plugin
+	PLUGIN_EXPORT void setPluginID(int32_t id);
```

Comment:
The plugin author is seeing this function by name "setPluginID" if I see a function that starts with "set" then I automatically think it's something I can call, which is not the case here.
It is essentially a eventhandler for when the mumble client notifies us about the id it assigned to us.
Also does this need to be signed int?

Reply:
Yes you are right. The naming is a bit misleading. Do you have a good suggestion for a replacement`Maybe something like `takePluginID`, `registerPluginID` or `acceptPluginID`? I think the latter on is quite descriptive...

> Also does this need to be signed int?
No I guess we could get along with unsigned integers quite nicely. Will change accordingly.

Reply:
registerPluginID sounds good, same for the API functions above I'd say.